daemon/graphdriver/devmapper/devmapper_test.go

```diff
@@ -40,3 +42,40 @@ func TestDevmapperCreateSnap(t *testing.T) {
 func TestDevmapperTeardown(t *testing.T) {
 	graphtest.PutDriver(t)
 }
+
+func TestDevmapperReduceLoopBackSize(t *testing.T) {
+	tenMB := int64(10 * 1024 * 1024)
+	testChangeLoopBackSize(t, -tenMB, defaultDataLoopbackSize, defaultMetaDataLoopbackSize)
+}
+
+func TestDevmapperIncreaseLoopBackSize(t *testing.T) {
+	tenMB := int64(10 * 1024 * 1024)
+	testChangeLoopBackSize(t, tenMB, defaultDataLoopbackSize+tenMB, defaultMetaDataLoopbackSize+tenMB)
+}
+
+func testChangeLoopBackSize(t *testing.T, delta, expectDataSize, expectMetaDataSize int64) {
+	driver := graphtest.GetDriver(t, "devicemapper").(*graphtest.Driver).Driver.(graphdriver.NaiveDiffDriver).ProtoDriver.(*Driver)
+	defer graphtest.PutDriver(t)
+	// make sure data or metadata loopback size are the default size
+	if s := driver.DeviceSet.Status(); s.Data.Total != uint64(defaultDataLoopbackSize) || s.Metadata.Total != uint64(defaultMetaDataLoopbackSize) {
+		t.Fatalf("data or metadata loop back size is incorrect")
+	}
+	if err := driver.Cleanup(); err != nil {
+		t.Fatal(err)
+	}
+	//Reload
+	d, err := Init(driver.home, []string{
+		fmt.Sprintf("dm.loopdatasize=%d", defaultDataLoopbackSize+delta),
+		fmt.Sprintf("dm.loopmetadatasize=%d", defaultMetaDataLoopbackSize+delta),
+	})
+	if err != nil {
+		t.Fatalf("error creating devicemapper driver: %v", err)
+	}
+	driver = d.(graphdriver.NaiveDiffDriver).ProtoDriver.(*Driver)
```

Comment:
Is this the whole reason for exporting the `NaiveDiffDriver`?

Then export it, but make the init function something like
```golang
type NaiveDiffDriver struct {
	ProtoDriver
}
func NewNaiveDiffDriver(driver ProtoDriver) Driver {
     return &NaiveDiffDriver{ProtoDriver: driver}
}
```

So you can have your exported type for type assertion, and still retain the rigidity.